lib/qa/authorities/geonames.rb

```diff
@@ -16,7 +16,23 @@ def search(q)
         Rails.logger.error 'Questioning Authority tried to call geonames, but no username was set'
         return []
       end
-      parse_authority_response(json(build_query_url(q)))
+
+      # CHECK: Making a check that if an id number or url is enter instead of city name
+      if q.to_s.match(/^[0-9]*$/)
+        parse_response_with_id(json(build_id_url(q)))
+      elsif q.to_s.match(%r{^https?:\/\/www\.geonames\.org})
```

Comment:
I'm not sure what's different in OD2, but you don't need to escape the `//` part if the regex is in a `%r{}` block apparently
That's `Style/RedundantRegexpEscape`